File: backend/app/routers/agent.py
from __future__ import annotations

from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session

from app.agent.graph import run_agent_chat
from app.core import config
from app.core.database import get_db
from app.core.response import success_response
from app.core.security import get_current_user
from app.models.user import User
from app.schemas.agent import AgentChatRequest, AgentChatResponse

router = APIRouter(prefix="/agent", tags=["agent"])


@router.post("/chat")
def chat_with_agent(
    payload: AgentChatRequest,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
) -> dict:
    if not config.DEEPSEEK_API_KEY:
        raise HTTPException(
            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
            detail="AI 服务未配置，请先设置 DEEPSEEK_API_KEY",
        )

    data = run_agent_chat(
        db=db,
        current_user=current_user,
        message=payload.message,
        history=payload.history,
    )
    return success_response(data=AgentChatResponse(**data).model_dump())

File: mobile/src/api/agent.ts
import type { AgentChatMessage, AgentChatResponse, ApiResponse } from '../types';
import client, { unwrap } from './client';

/** 发送 AI 记账助手消息(带历史上下文)。 */
export const sendAgentMessage = async (
  message: string,
  history: AgentChatMessage[],
): Promise<AgentChatResponse> => {
  const response = await client.post<ApiResponse<AgentChatResponse>>(
    '/agent/chat',
    { message, history },
  );
  return unwrap(response.data);
};
